olmo/util.py

```diff
@@ -841,12 +841,28 @@ def get_bytes_range(self, index: int, length: int) -> bytes:
         return response["Body"].read()
 
 
-def flatten_dict(dictionary, parent_key="", separator="."):
+def flatten_dict(dictionary, parent_key="", separator=".", include_lists=False):
+    """
+    Flatten a nested dictionary into a single-level dictionary.
+
+    Args:
+        dictionary (dict): The nested dictionary to be flattened.
+        parent_key (str, optional): The parent key to be prepended to the keys of the flattened dictionary. Defaults to "".
+        separator (str, optional): The separator to be used between the parent key and the keys of the flattened dictionary. Defaults to ".".
+        include_lists (bool, optional): Whether to convert lists to dictionaries with integer keys. Defaults to False.
```

Comment:
Do we ever want to turn this off?

Reply:
not for now, but seems fine to extend this in case someone wants to add different logic for dealing w list-valued config params?